src/expr/src/scalar/func/impls/timestamp.rs

```diff
@@ -121,19 +121,22 @@ impl fmt::Display for CastTimestampToTimestampTz {
 #[derive(
     Arbitrary, Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, MzReflect,
 )]
-pub struct CastTimestampToTimestamp {
+pub struct AdjustTimestampPrecision {
     pub from: Option<TimestampPrecision>,
     pub to: Option<TimestampPrecision>,
 }
 
-impl<'a> EagerUnaryFunc<'a> for CastTimestampToTimestamp {
+impl<'a> EagerUnaryFunc<'a> for AdjustTimestampPrecision {
     type Input = CheckedTimestamp<NaiveDateTime>;
     type Output = Result<CheckedTimestamp<NaiveDateTime>, EvalError>;
 
     fn call(
         &self,
         a: CheckedTimestamp<NaiveDateTime>,
     ) -> Result<CheckedTimestamp<NaiveDateTime>, EvalError> {
+        if self.to == self.from {
```

Comment:
I don't think evaluating this is quite right because `to` can never equal `from`, otherwise we have a panic.

Reply:
Do you mean we should panic here instead? Could that break things?

Reply:
Yes, the clearer thing to do would be to put the call to `mz_ore::soft_assert` in this spot rather than in `preserves_uniqueness`.

Panicking in these kinds of situations is expressly the purpose of `soft_assert`--it will panic in dev environments so will catch any bugs we introduce in planning, but is removed from production builds so will not harm customers if they somehow trip over it.

Reply:
Ah I see. Let me update then.

Reply:
Done!